Missing WebRTC stats in Safari

youennf opened the pull request:
#### 7b5229864801f85e858872ce7356e758ce32a244
<pre>
Missing WebRTC stats in Safari
<a href="https://bugs.webkit.org/show_bug.cgi?id=267696">https://bugs.webkit.org/show_bug.cgi?id=267696</a>
<a href="https://rdar.apple.com/121594743">rdar://121594743</a>

Reviewed by Eric Carlson.

Update datachannelid to dataChannelIdentifier name, as per spec.
Expose more RTC stats by exposing libwebrtc backend gathered values.

* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-mandatory-getStats.https-expected.txt:
* LayoutTests/webrtc/candidate-stats.html:
* LayoutTests/webrtc/datachannel/datachannel-stats.html:
* LayoutTests/webrtc/video-stats-expected.txt:
* LayoutTests/webrtc/video-stats.html:
* Source/WebCore/Modules/mediastream/RTCStatsReport.h:
* Source/WebCore/Modules/mediastream/RTCStatsReport.idl:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCStatsCollector.cpp:
(WebCore::fillInboundRtpStreamStats):
(WebCore::fillOutboundRtpStreamStats):
(WebCore::fillRTCDataChannelStats):
(WebCore::fillRTCIceCandidateStats):

Canonical link: <a href="https://commits.webkit.org/273643@main">https://commits.webkit.org/273643@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f34ccd83e024ec25729dee5ad27d1ce99473bac3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36142 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15089 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38344 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38863 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32493 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37364 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17527 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12117 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31185 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36697 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12743 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32065 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11171 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11195 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32248 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40109 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32808 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32613 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37134 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11391 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9276 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35220 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13115 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/31901 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8218 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11868 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12228 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Diego-Perez-Botero commented on Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCStatsCollector.cpp:
```diff
@@ -258,9 +265,8 @@ static inline void fillOutboundRtpStreamStats(RTCStatsReport::OutboundRtpStreamS
-    // if (rtcStats.ssrc.is_defined())
-    //     stats.rtxSsrc = *rtcStats.ssrc;
+    if (rtcStats.ssrc.is_defined())
+        stats.rtxSsrc = *rtcStats.ssrc;
```
Should be modifying "stats.ssrc" instead of "stats.rtxSsrc" here

fippo replied:
This one is simply not needed, rtxSsrc is set below while ssrc is already set by another method.

Diego-Perez-Botero replied:
You're right 🫣

youennf committed:
Missing WebRTC stats in Safari
https://bugs.webkit.org/show_bug.cgi?id=267696
rdar://121594743

Reviewed by Eric Carlson.

Update datachannelid to dataChannelIdentifier name, as per spec.
Expose more RTC stats by exposing libwebrtc backend gathered values.

* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-mandatory-getStats.https-expected.txt:
* LayoutTests/webrtc/candidate-stats.html:
* LayoutTests/webrtc/datachannel/datachannel-stats.html:
* LayoutTests/webrtc/video-stats-expected.txt:
* LayoutTests/webrtc/video-stats.html:
* Source/WebCore/Modules/mediastream/RTCStatsReport.h:
* Source/WebCore/Modules/mediastream/RTCStatsReport.idl:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCStatsCollector.cpp:
(WebCore::fillInboundRtpStreamStats):
(WebCore::fillOutboundRtpStreamStats):
(WebCore::fillRTCDataChannelStats):
(WebCore::fillRTCIceCandidateStats):

Canonical link: https://commits.webkit.org/273643@main
